pkg/sloop/storemanager/stats.go

```diff
@@ -16,21 +16,27 @@ const vlogExt = ".vlog" // value log data
 const sstExt = ".sst"   // LSM data
 
 var (
-	metricStoreSizeOnDiskMb   = promauto.NewGauge(prometheus.GaugeOpts{Name: "sloop_store_sizeondiskmb"})
-	metricBadgerKeys          = promauto.NewGaugeVec(prometheus.GaugeOpts{Name: "sloop_badger_keys"}, []string{"level"})
-	metricBadgerTables        = promauto.NewGaugeVec(prometheus.GaugeOpts{Name: "sloop_badger_tables"}, []string{"level"})
-	metricBadgerLsmFileCount  = promauto.NewGauge(prometheus.GaugeOpts{Name: "sloop_badger_lsmfilecount"})
-	metricBadgerLsmSizeMb     = promauto.NewGauge(prometheus.GaugeOpts{Name: "sloop_badger_lsmsizemb"})
-	metricBadgerVLogFileCount = promauto.NewGauge(prometheus.GaugeOpts{Name: "sloop_badger_vlogfilecount"})
-	metricBadgerVLogSizeMb    = promauto.NewGauge(prometheus.GaugeOpts{Name: "sloop_badger_vlogsizemb"})
+	metricStoreSizeOnDiskMb          = promauto.NewGauge(prometheus.GaugeOpts{Name: "sloop_store_sizeondiskmb"})
+	metricBadgerKeys                 = promauto.NewGaugeVec(prometheus.GaugeOpts{Name: "sloop_badger_keys"}, []string{"level"})
+	metricBadgerTables               = promauto.NewGaugeVec(prometheus.GaugeOpts{Name: "sloop_badger_tables"}, []string{"level"})
+	metricBadgerLsmFileCount         = promauto.NewGauge(prometheus.GaugeOpts{Name: "sloop_badger_lsmfilecount"})
+	metricBadgerLsmSizeMb            = promauto.NewGauge(prometheus.GaugeOpts{Name: "sloop_badger_lsmsizemb"})
+	metricBadgerVLogFileCount        = promauto.NewGauge(prometheus.GaugeOpts{Name: "sloop_badger_vlogfilecount"})
+	metricBadgerVLogSizeMb           = promauto.NewGauge(prometheus.GaugeOpts{Name: "sloop_badger_vlogsizemb"})
+	metricCleanedStoreSizeOnDiskMb   = promauto.NewGauge(prometheus.GaugeOpts{Name: "sloop_garbage_collected_sizeondiskmb"})
+	metricCleanedBadgerKeys          = promauto.NewGaugeVec(prometheus.GaugeOpts{Name: "sloop_garbage_collected_badger_keys"}, []string{"level"})
+	metricCleanedBadgerLsmFileCount  = promauto.NewGauge(prometheus.GaugeOpts{Name: "sloop_garbage_collected_badger_lsmfilecount"})
+	metricCleanedBadgerLsmSizeMb     = promauto.NewGauge(prometheus.GaugeOpts{Name: "sloop_garbage_collected_badger_lsmsizemb"})
+	metricCleanedBadgerVLogFileCount = promauto.NewGauge(prometheus.GaugeOpts{Name: "sloop_garbage_collected_badger_vlogfilecount"})
+	metricCleanedBadgerVLogSizeMb    = promauto.NewGauge(prometheus.GaugeOpts{Name: "sloop_garbage_collected_badger_vlogsizemb"})
 )
 
 type storeStats struct {
 	timestamp         time.Time
-	DiskSizeBytes     uint64
-	DiskLsmBytes      uint64
+	DiskSizeBytes     int64
```

Comment:
Why is this changing?

Reply:
The delta of size is a signed number and the delta is calculated suing the difference between  value before and after GC. Hence the type of the variable had to be changed.